EC_finetune/__main__.py

```diff
@@ -281,6 +303,22 @@ def main():
     if args.adapter_freeze_override:
         args.freeze_adapters = True
 
+    if args.sender_freeze_override:
```

Comment:
All of this logic seems like over-kill to me.  I think you can simplify all of this just in the `add_argument` call, by setting `dest=` and `default=` to each of those, e.g. 
```
parser.add_argument(
        '--sender_freeze_override',
        action='store_true',
        help="Flag to trigger sender freezing (overriding config)",
       dest="freeze_sender", 
       default=False
    )
```
I'm 95% about this, but check the argparse docs to confirm :)

Reply:
I just checked the argparse docs, so we basically parse these arguments first and then we load the yml files into the args object. Then wouldn't the argument/add_argument thing above be overwrriten?